Microsoft.HealthVault.Client.NetFramework/NetFrameworkSecretStore.cs

```
@@ -0,0 +1,113 @@
+
+        public async Task WriteAsync(string key, string contents)
+        {
+            try
+            {
+                string filePath = Path.Combine(StoreFolder, key);
+
+                EnsureDirectory();
+
+                byte[] contentBytes = Encoding.UTF8.GetBytes(contents);
+                byte[] encryptedContentBytes = ProtectedData.Protect(contentBytes, Salt, DataProtectionScope.CurrentUser);
+
+                using (FileStream fileStream = new FileStream(filePath, FileMode.Create, FileAccess.Write))
+                {
+                    await fileStream.WriteAsync(encryptedContentBytes, 0, encryptedContentBytes.Length).ConfigureAwait(false);
+                }
+            }
+            catch (Exception e) when (!(e is IOException))
+            {
+                throw new IOException(string.Format(ClientResources.FileAccessErrorMessage, ClientResources.FileAccessActionWrite, key), e);
+            }
+        }
+
+        public async Task<string> ReadAsync(string key)
+        {
+            try
+            {
+                string filePath = Path.Combine(StoreFolder, key);
+
+                using (FileStream fileStream = new FileStream(filePath, FileMode.Open, FileAccess.Read))
+                {
+                    byte[] encryptedContentBytes = new byte[fileStream.Length];
+                    await fileStream.ReadAsync(encryptedContentBytes, 0, (int)fileStream.Length).ConfigureAwait(false);
+                    byte[] contentBytes = ProtectedData.Unprotect(encryptedContentBytes, Salt, DataProtectionScope.CurrentUser);
+
+                    return Encoding.UTF8.GetString(contentBytes);
+                }
+            }
+            catch (FileNotFoundException)
+            {
+                return null;
+            }
+            catch (DirectoryNotFoundException)
+            {
+                return null;
+            }
```

Why hide the error details behind a null? Should we let the caller make the decision?